ecommers/lib/core/app_services/dynamic_link_service.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:ecommers/core/models/data_models/index.dart';
+import 'package:ecommers/ui/utils/index.dart';
+import 'package:firebase_dynamic_links/firebase_dynamic_links.dart';
+import 'package:ecommers/shared/dependency_service.dart';
+import 'package:ecommers/core/common/index.dart';
+
+class DynamicLinkService {
+  Future handleDynamicLinks() async {
+    final PendingDynamicLinkData data =
+        await FirebaseDynamicLinks.instance.getInitialLink();
+
+    _handleDeepLink(data);
+
+    FirebaseDynamicLinks.instance.onLink(
+        onSuccess: (PendingDynamicLinkData dynamicLink) async {
+      _handleDeepLink(dynamicLink);
+    }, onError: (OnLinkErrorException e) async {
+      logger.ex(e);
+    });
+  }
+
+  void _handleDeepLink(PendingDynamicLinkData data) {
+    final Uri deepLink = data?.link;
+
+    if (deepLink != null && membershipService.isNotExpired) {
+      final productId = int.parse(deepLink.queryParameters['id']);
+      _fetchProductByIdAndNavigation(productId);
+    }
+  }
+
+  Future _fetchProductByIdAndNavigation(int id) async {
+    final Product product = await productService.fetchProductById(id);
```

Comment:
remove method, and move body to  if (deepLink != null && membershipService.isNotExpired) {
      final productId = int.parse(deepLink.queryParameters['id']);
      _fetchProductByIdAndNavigation(productId);
    }

Reply:
removed method